tests/unit/llms/providers/test_watsonx.py

```diff
@@ -40,3 +43,74 @@ def test_basic_interface(provider_config):
     llm = watsonx.load()
     assert isinstance(llm, WatsonxLLM)
     assert watsonx.default_params
+
+
+@patch("ols.src.llms.providers.watsonx.WatsonxLLM", new=WatsonxLLM())
+def test_params_handling(provider_config):
+    """Test that not allowed parameters are removed before model init."""
+    config.init_empty_config()  # needed for checking the config.dev_config.llm_params
+
+    # first three parameters should be removed before model init
+    # rest need to stay
```

Comment:
guessing this comment was copy+pasted above because it looks like it's accurate here.  but if you're going to setup all these parameters, i'd expect the test to check all of them (ensure they are either present, or not present, but do something with them. Setting them and then not checking/using them in the test is strange and a bit confusing in terms of understanding what is being tested/what behavior is expected)

Reply:
updated there too @bparees